pkg/connector/sso_group.go

```diff
@@ -155,6 +169,81 @@ func ssoGroupBuilder(region string, ssoClient *awsSsoAdmin.Client, identityStore
 	}
 }
 
+func (g *ssoGroupResourceType) Grant(ctx context.Context, principal *v2.Resource, entitlement *v2.Entitlement) (annotations.Annotations, error) {
+	if principal.Id.ResourceType != resourceTypeSSOUser.Id {
+		return nil, errors.New("baton-aws: only sso users can be added to a sso group")
+	}
+
+	entitlementAnnotations := annotations.Annotations(entitlement.Resource.Annotations)
+	groupTrait := &v2.GroupTrait{}
+	ok, err := entitlementAnnotations.Pick(groupTrait)
+	if err != nil {
+		return nil, fmt.Errorf("baton-aws: error unmarshalling entitlement group trait: %w", err)
+	}
+	if !ok {
+		return nil, errors.New("baton-aws: entitlement missing group trait")
+	}
+	groupID, ok := resourceSdk.GetProfileStringValue(groupTrait.Profile, "aws_group_id")
+	if !ok {
+		return nil, errors.New("baton-aws: entitlement group trait missing aws_group_id")
+	}
+
+	userAnnotations := annotations.Annotations(principal.Annotations)
+	userTrait := &v2.UserTrait{}
+	ok, err = userAnnotations.Pick(userTrait)
+	if err != nil {
+		return nil, fmt.Errorf("baton-aws: error unmarshalling user trait: %w", err)
+	}
+	if !ok {
+		return nil, errors.New("baton-aws: principal missing user trait")
+	}
+	userID, ok := resourceSdk.GetProfileStringValue(userTrait.Profile, "aws_user_id")
+	if !ok {
+		return nil, errors.New("baton-aws: principal user trait missing aws_user_id")
+	}
+
+	input := &awsIdentityStore.CreateGroupMembershipInput{
+		GroupId:         awsSdk.String(groupID),
+		IdentityStoreId: g.identityInstance.IdentityStoreId,
+		MemberId:        &awsIdentityStoreTypes.MemberIdMemberUserId{Value: userID},
+	}
+
+	if _, err := g.identityStoreClient.CreateGroupMembership(ctx, input); err != nil {
+		return nil, fmt.Errorf("baton-aws: error adding sso user to sso group: %w", err)
+	}
+
+	return nil, nil
+}
+func (g *ssoGroupResourceType) Revoke(ctx context.Context, grant *v2.Grant) (annotations.Annotations, error) {
+	if grant.Principal.Id.ResourceType != resourceTypeSSOUser.Id {
+		return nil, errors.New("baton-aws: only sso users can be removed from sso group")
+	}
+
+	grantAnnos := annotations.Annotations(grant.Annotations)
+	meta := &structpb.Struct{}
+	ok, err := grantAnnos.Pick(meta)
+	if err != nil {
+		return nil, fmt.Errorf("baton-aws: error unmarshalling grant metadata: %w", err)
+	}
+	if !ok {
+		return nil, errors.New("baton-aws: grant missing metadata")
+	}
+
+	membershipID, ok := resourceSdk.GetProfileStringValue(meta, MembershipIDMetadataKey)
```

Comment:
when making the grant, should we store this memebership ID as the grant ID? That'd let us avoid having to look it up in the grant metadata object?